crates/engine/src/engine/cache/warming.rs

```diff
@@ -0,0 +1,89 @@
+use std::sync::Arc;
+
+use runtime::operation_cache::OperationCache;
+
+use super::{DocumentKey, Key};
+
+use crate::{
+    engine::{EarlyHttpContext, Engine, ResponseFormat, Runtime},
+    prepare::PrepareContext,
+    CachedOperation,
+};
+
+pub struct OperationForWarming {
```

Comment:
Would just have named this something like `RawOperation` and included the struct directly into the `CachedOperation`. You're adding exactly the same fields on both side as far as I've understood

Reply:
Not a bad suggestion, but I don't want to wait on CI again.  Will do in a future PR if I remember.